File: backend/gatorsched_api/schemas/auth/create_employee_account.py
from pydantic import BaseModel, EmailStr


class CreateEmployeeAccountRequest(BaseModel):
    name: str
    email: EmailStr
    password: str
    phone: str | None = None
    avatarUrl: str | None = None
    isActive: bool
    roleName: str


class CreateEmployeeAccountResponse(BaseModel):
    success: bool

File: frontend/src/services/auth/createEmployeeAccount.ts
import { API_BASE } from '../apiConfig';

export type CreateEmployeeRequest = {
  name: string;
  email: string;
  password: string;
  phone: string | null;
  avatarUrl: string | null;
  roleName: string;
};

export type CreateEmployeeResponse = {
  success: boolean;
};

export async function createEmployeeAccount(
  payload: CreateEmployeeRequest,
): Promise<CreateEmployeeResponse> {
  const res = await fetch(`${API_BASE}/auth/create-employee-account`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(payload),
  });

  if (!res.ok) {
    throw new Error('Failed to create employee account');
  }

  return await res.json();
}
